From 30205c8268d0a82c499b129037be1741a21f95d8 Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Sat, 1 Nov 2025 21:32:44 +0100
Subject: [PATCH] Disable MOBILEAP_PRIORITIZE_TRAFFIC support

Rebased for the S24U BP4A semwifi-service: the traffic priority status
now lives in SemWifiApBroadcastReceiver.getTrafficPriorityStatus().

---
diff --git a/smali/com/samsung/android/server/wifi/SemWifiManagerProxy.smali b/smali/com/samsung/android/server/wifi/SemWifiManagerProxy.smali
index 47fae71..062e8a2 100644
--- a/smali/com/samsung/android/server/wifi/SemWifiManagerProxy.smali
+++ b/smali/com/samsung/android/server/wifi/SemWifiManagerProxy.smali
@@ -2285,7 +2285,7 @@
 .method public isWifiApSmartPrioritySupported()Z
     .locals 0
 
-    const/4 p0, 0x1
+    const/4 p0, 0x0
 
     return p0
 .end method
diff --git a/smali/com/samsung/android/server/wifi/ap/SemSettingsSoftResetHandler.smali b/smali/com/samsung/android/server/wifi/ap/SemSettingsSoftResetHandler.smali
index 3c8e23d..208689d 100644
--- a/smali/com/samsung/android/server/wifi/ap/SemSettingsSoftResetHandler.smali
+++ b/smali/com/samsung/android/server/wifi/ap/SemSettingsSoftResetHandler.smali
@@ -198,24 +198,10 @@
 .method public resetHotspotSettings()V
     .locals 6
 
-    const-string v0, "disabling traffic priority"
-
     const-string v1, "SemSettingsSoftResetHandler"
 
-    invoke-static {v1, v0}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/samsung/android/server/wifi/ap/SemSettingsSoftResetHandler;->mContext:Landroid/content/Context;
-
-    invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
-
-    move-result-object v0
-
-    const-string v2, "wifi_ap_smart_priority_enable"
-
     const/4 v3, 0x0
 
-    invoke-static {v0, v2, v3}, Landroid/provider/Settings$Secure;->putInt(Landroid/content/ContentResolver;Ljava/lang/String;I)Z
-
     sget-object v0, Lcom/samsung/android/server/wifi/ap/SemSettingsSoftResetHandler;->CONFIGOPBRANDINGFORMOBILEAP:Ljava/lang/String;
 
     const-string v2, "TMO"
diff --git a/smali/com/samsung/android/server/wifi/ap/SemWifiApBroadcastReceiver.smali b/smali/com/samsung/android/server/wifi/ap/SemWifiApBroadcastReceiver.smali
index b5cee9a..7f81ccb 100644
--- a/smali/com/samsung/android/server/wifi/ap/SemWifiApBroadcastReceiver.smali
+++ b/smali/com/samsung/android/server/wifi/ap/SemWifiApBroadcastReceiver.smali
@@ -1480,63 +1480,9 @@
 .end method
 
 .method public final getTrafficPriorityStatus()Ljava/lang/String;
-    .locals 2
-
-    invoke-static {}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getInstance()Lcom/samsung/android/server/wifi/SemWifiInjector;
-
-    move-result-object p0
-
-    invoke-virtual {p0}, Lcom/samsung/android/server/wifi/SemWifiInjector;->getSemWifiApTrafficPriority()Lcom/samsung/android/server/wifi/ap/smartprio/SemWifiApTrafficPriority;
-
-    move-result-object p0
-
-    if-eqz p0, :cond_2
-
-    invoke-virtual {p0}, Lcom/samsung/android/server/wifi/ap/smartprio/SemWifiApTrafficPriority;->getmIsPriorityFeatureEnableDuringSession()Z
-
-    move-result v0
-
-    if-eqz v0, :cond_2
-
-    invoke-virtual {p0}, Lcom/samsung/android/server/wifi/ap/smartprio/SemWifiApTrafficPriority;->getNRTTrafficEnable()Z
-
-    move-result v0
-
-    if-eqz v0, :cond_1
-
-    invoke-virtual {p0}, Lcom/samsung/android/server/wifi/ap/smartprio/SemWifiApTrafficPriority;->getRTTrafficEnable()Z
-
-    move-result p0
-
-    if-eqz p0, :cond_0
-
-    const-string p0, "tp_rt"
-
-    goto :goto_0
-
-    :cond_0
-    const-string p0, "tp_nrt"
-
-    goto :goto_0
-
-    :cond_1
-    const-string p0, "tp_on_below_cutoff"
-
-    goto :goto_0
-
-    :cond_2
-    const-string p0, "tp_off"
-
-    :goto_0
-    const-string v0, "handleWifiApState: "
-
-    invoke-virtual {v0, p0}, Ljava/lang/String;->concat(Ljava/lang/String;)Ljava/lang/String;
-
-    move-result-object v0
-
-    const-string v1, "SemWifiApBroadcastReceiver"
+    .locals 0
 
-    invoke-static {v1, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+    const-string p0, "not_support"
 
     return-object p0
 .end method
-- 
2.55.0
